nilearn/mass_univariate/_utils.py

```diff
@@ -65,50 +65,68 @@ def _calculate_tfce(
     """
     tfce_4d = np.zeros_like(arr4d)
 
+    # For each passed t map
     for i_regressor in range(arr4d.shape[3]):
         arr3d = arr4d[..., i_regressor]
+
+        # Get signs / threshs
         if two_sided_test:
             signs = [-1, 1]
-
-            # Get the maximum statistic in the map
             max_score = np.max(np.abs(arr3d))
         else:
             signs = [1]
-
-            # Get the maximum statistic in the map
             max_score = np.max(arr3d)
 
         if dh == 'auto':
             step = max_score / 100
         else:
             step = dh
 
-        for score_thresh in np.arange(step, max_score + step, step):
-            for sign in signs:
-                temp_arr3d = arr3d * sign
+        # Set based on determined step size
+        score_threshs = np.arange(step, max_score + step, step)
+
+        # If we apply the sign first...
+        for sign in signs:
+
+            # Then init a temp array, we can re-use the
+            # same array and save a little bit of time
+            temp_arr3d = arr3d.copy() * sign
```

Comment:
Why do you add the `copy` method here?

Reply:
Just to make it a little more clear that we are working with a copy of arr3d. Though as you point out it is not strictly necessary as "temp_arr3d = arr3d * sign" would also create a copy

Reply:
I think the copy will slow things down very slightly, and since it's not necessary I think it would be good to drop it.

Reply:
Sure no worries, I just feel like sometimes the behavior isn't super clear, e.g., "temp_arr3d = arr3d" alone just sets temp_arr3d as a reference to the same array, but I'm always down to chase the microsecond gains.